core/src/main/scala/scalaz/Foldable.scala

```diff
@@ -325,6 +325,9 @@ trait Foldable[F[_]]  { self =>
         else seen
     }.reverse
 
+  def distinctBy[A, B: Equal](fa: F[A])(f: A => B): IList[A] =
+    distinctE(fa)(Equal.equalBy(f))
```

Comment:
alternatively `distinctE` could be implemented in term of `distinctBy` so that the explicit `Equal` would not be required.